soc/nxp/rw/CMakeLists.txt

```diff
@@ -0,0 +1,13 @@
+# Copyright 2022-2024 NXP
+#
+# SPDX-License-Identifier: Apache-2.0
+
+zephyr_sources(
+  soc.c
+  flexspi_clock_setup.c
```

Comment:
Can flexspi_clock_setup.c be added conditionally based on the component that needs it.